### ISSUE TITLE
Missing `convert(Vector, ::SubArray)`

### ISSUE THREAD

**Issue by cstjean** [JuliaLang/julia#16519]
On 0.4:

```julia
convert(Vector, slice(collect(1:10), 2:4))
```

is not defined, while `convert(Array, slice(...))` is. I don't have 0.5 yet - please close if it's fixed there.

**Comment by nalimilan** [comment]
Doesn't work on 0.5 either.

**Comment by Evizero** [comment]
This may be a stupid question, but should `convert` work like that?

This code does work (v0.4):

```julia
convert(Vector{Int}, sub(collect(1:10), 1:2))
```
```
2-element Array{Int64,1}:
 1
 2
```

So does 

```julia
convert(Vector{Float64}, sub(collect(1:10), 1:2))
```
```
2-element Array{Float64,1}:
 1.0
 2.0
```

Then again, if it works with `convert(Array, ...)` it would make sense to also work with `Vector`

**Comment by Sacha0** [comment]
> Then again, if it works with convert(Array, ...) it would make sense to also work with Vector

Having `convert(Vector, view(collect(1:10), 1:2))` work would be consistent with the direction in #17066 / related threads. Adding
```julia
convert{T}(::Type{Vector}, A::AbstractVector{T}) = convert(Vector{T},  A)
```
just after [this line](https://github.com/JuliaLang/julia/blob/f9fb4dae619ce2f9e7b3aa466b27e4c4f6aff11b/base/abstractarray.jl#L626) should do the trick? And if so, might as well add
```julia
convert{T}(::Type{Matrix}, A::AbstractMatrix{T}) = convert(Matrix{T}, A)
```
in the process? Or perhaps a more generic solution exists? Best!

**Comment by Evizero** [comment]
it can't go in abstractarray.jl because it will make compilation fail. array.jl seems like the best place. I am investigating

**Comment by Sacha0** [comment]
> it can't go in abstractarray.jl because it will make compilation fail. array.jl seems like the best place. I am investigating

Good call, array.jl sounds like the best place. Best!

**Comment by nalimilan** [comment]
Fixed in master by #17848. Waiting for backport to 0.5.
